formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -279,6 +279,10 @@ internal class FormulaManagerImpl<Input, State, Output>(
      * @return True if we need to re-evaluate.
      */
     private fun postEvaluation(evaluationId: Long): Boolean {
+        if (isEvaluationNeeded(evaluationId)) {
```

Comment:
This fixes an issue where a child state might have changed during evaluation and wasn't picked up by us. This ensures we re-evaluate again.